### DISCUSSION TITLE
Configuration and log files where they are in OSX ARM?

### DISCUSSION THREAD

**Discussion by IU2NUB-Roberto** [foldynl/QLog#357] [General]
Good morning
I'm used to having all my files on a NAS so I can access them from any PC/Mac I have online
Normally, I create the folder in the standard location, copy it to the NAS and then create a symbolic link in the original location
This allows me to overcome the exclusive opening of some programs and always have everything synchronized
The problem is that I can't find where the QLOG log and configuration files are in my macOSX ARM m1Pro
Can anyone give me an idea?
Thank you
Roberto

**Comment by foldynl** [comment]
Unfortunately, this approach is not supported by QLog. In the other word, it is not possible to share the qlog db and configuration via a shared storage.

**Reply by IU2NUB-Roberto** [reply]
Thanks, but it's a real shame, the software is interesting
I will wait and continue to use other loggers with regret
Thank you
roberto

**Comment by foldynl** [comment]
nothing is perfect. no one wants to contribute, so the progress is slow

**Comment by IU2NUB-Roberto** [comment]
I can understand and I thank you for the work done
If I were a few years younger I might think of helping out but now I'm too
old for that
 I am sorry
Thank you
roberto


Il giorno gio 25 apr 2024 alle 14:49 Ladislav ***@***.***> ha
scritto:

**Reply by foldynl** [reply]
No problem, Reberto. cross-platform synchronization is on the TODO list long time. But due to the internal architecture of QLog it is not easy to implement.

**Comment by IU2NUB-Roberto** [comment]
But can you tell me where are the files (log - config ecc) in apple architecture?
Roberto

**Reply by foldynl** [reply]
I have no MacOS therefore I'm just guessing. Based on QT Spec, DB file should be in ~/Library/Application Support/QLog.

Configuration should be placed $HOME/Library/Preferences/.... not sure what file.name. 

Configurations are platform depended !!!

**Comment by IU2NUB-Roberto** [comment]
ok
Thank you I'm not at home now
I'll look for them as soon as I get back
roberto


Il giorno gio 25 apr 2024 alle 15:16 Ladislav ***@***.***> ha
scritto:

**Comment by IU2NUB-Roberto** [comment]
Good morning Ladislav
I searched for qlog files and the result is:
the configuration file is in: $HOME/Library/Preferences/com.hamradio.qlog.plist

but I absolutely couldn't find the log file and the accessory files like cty.csv etc
I tried to read the configuration file but it does not contain any reference to the positioning of the files.

I tried installing a copy of Qlog on a w11 machine (virtual via parallels) and all the files are placed in:
C:\Users\xxxxxxxx\AppData\Local\hamradio\QLog

If I may allow myself a small request, it would be to be certain where the files are in macosx Sonoma ARM
Not finding them anywhere, I imagine they are in some hidden folder which I, however, searched for by forcing the display (Shift + command + dot)
This also makes me think of an impossibility of recovery from time machines or other backup systems which makes Qlog "dangerous" to use in my favorite environment since if something happens I can't recover the data (except doing an export every time that I use it).

Sorry for the length and frankness, but after 43 years of computing I think it is correct to be clear and honest.
Thanks again for the wonderful work you are doing.
I will continue to monitor Qlog, I really like it
Roberto

**Reply by foldynl** [reply]
QLog is based on QT.  To be able to develop cross-platform app, everything must be controlled by this library. I don't have access to a Mac myself, the Mac version is only the result of a compilation using Github Actions. On this platform, I only depend on user reports and volunteers to help me pull this together.

I would be surprised that QT uses any application folder that are outside of your HOME.

DB and all downloaded objects, Internal QLog's backups are stored into the ``QStandardPaths::AppLocalDataLocation`` which is defined:

Mac: ``"~/Library/Application Support/<APPNAME>", "/Library/Application Support/<APPNAME>". "<APPDIR>/../Resources"``
Win: ``"C:/Users/<USER>/AppData/Local/<APPNAME>", "C:/ProgramData/<APPNAME>", "<APPDIR>", "<APPDIR>/data", "<APPDIR>/data/<APPNAME>"``
Linux: ``"~/.local/share/<APPNAME>", "/usr/local/share/<APPNAME>", "/usr/share/<APPNAME>"``

The first records is usually used.

>>Sorry for the length and frankness, but after 43 years of computing I think it is correct to be clear and honest.
Thanks again for the wonderful work you are doing.
I will continue to monitor Qlog, I really like it

I don't have a problem with your monitoring. It's your choice, but simply monitoring will  make NO progress.

**Comment by IU2NUB-Roberto** [comment]
"but simply monitoring will make NO progress”.

What do you mean by not making progress, what do you think I should do to make you progress?
roberto



> Il giorno 26 apr 2024, alle ore 10:52, Ladislav ***@***.***> ha scritto:
> 
> but simply monitoring will make NO progress.

**Reply by foldynl** [reply]
There are many tasks for Macos platform

you can:

1) Find out where DB file and backup are placed on MacOS
2) update WIKI  [Autobackup](https://github.com/foldynl/QLog/wiki/Autobackup) chapter
3) do a research, if the timemachine backups the directory. if not then suggest solution for MacOS which is compatible with the Windows and Linux platform

These tasks are related to your issue report. 

Then there are many other task related to Macos (e.g #269, Migration Github Actions Build Pipe to MacOS13 + testing, #269 also mentions problem with QLog on Apple Silicon chip (MacOS14); it seems that binary has to be signed. is there any alternative to signed  it without having to pay Apple Developer Subscription ? Does [Casks](https://docs.brew.sh/Adding-Software-to-Homebrew#casks) solve the problem with Apple Silicon? Should QLog be distributed via Casks. How to achieve it?

These and many other questions are open for this platform

**Reply by IU2NUB-Roberto** [reply]
Ladislav good morning,
I found, as you requested, the location of the Qlog files in the Mac environment, and I did all the tests you requested, but I am not able to modify the "Autobakup" tab in the wiki, I can read it but not modify it.
I'm also working on the other options you mentioned.
Kindly tell me how/what should I do.
Roberto
PS i have a gthub account

**Reply by foldynl** [reply]
Perfect Roberto. 

WIKI - just send me a text. I will include it to wiki. WIKI is github repository but it will be safe that you send me the text and I will updated it because currently, I have prepared a new WIKI for upcoming release 0.35.

if you have any QLog improvements connected to the source code, the preferred format is to send a Github Pull Request. But you can send "diff" by email. The format is not as strict as other open-source projects request.

>> I'm also working on the other options you mentioned.

it would be great if some steps could be taken in these areas.

**Reply by IU2NUB-Roberto** [reply]
ok Ladislav,
so let's get started, I will also insert some comments and/or "errors" that I found.
x WIKI (OSX Sonoma 14.4.1 Apple Silicon)
the location of the Qlog library is:
$HOME\library\Application support\hamradio\qlog

The location of the Qlog configuration file is:
$HOME\Library\Preferences\com.hamradio.qlog.plist
the problem is that this library is normally hidden and the user cannot see it.
To view it you need to enter the command:
shift-command+. (dot)
But even using Find on your Mac you can't find it.
The backup procedure via Time Machine saves it, but even in this case, browsing the backups, you can't see it.
I don't know if with the system you use during compilation it is possible to "move" this data to an easily accessible position, the alternative is to create a "button" that retrieves saved data (but I find it difficult from a time machine).

Other things found during use.
- In the OSX version the online map is not activated, while in Windows it is perfect.

- (maybe I haven't read the manual enough) but I would like it in the cluster:
a- being able to enter the username and password of the servers to which I am connected
b- being able to divide by spot origin (a spot from Oceania is of little use in Europe)
c- being able to decide which spots to see (only cw, digital, ssb or a mix of them)

-when you close Qlog, it creates, rightly, a backup *adx file always within the same directory.
Having done various tests I already find myself with 650mb of files
It would be helpful to plan to write these files to a place of the user's choice

For now I would say sufficient even if I already have other "problems" detected

Thank you
Roberto

**Reply by foldynl** [reply]
Many thanks. I will update the WIKI based on your findings. 

Other things: WIKI will answer many of your questions.
